code/modules/mob/dead/new_player/poll.dm

```diff
@@ -1,620 +1,502 @@
+	if(!length(voted_for) || poll.allow_revoting)
+		output += {"<form action='?src=[REF(src)]' method='POST'>
+		<input type='hidden' name='src' value='[REF(src)]'>
+		<input type='hidden' name='votepollref' value='[REF(poll)]'>
+		<input type='hidden' name='IRVdata' id='IRVdata'>
+		"}
+	output += "<div id='ballot' class='center'><b><center>Most Preferred</center></b><ol id='sortable' class='rankings' style='padding:0px'>"
+	for(var/o in prepared_options)
+		var/datum/poll_option/option = o
+		output += "<li optionref='[REF(option)]' class='ranking'><span class='grippy'></span> [option.text]</li>\n"
+	output += "</ol><b><center>Least Preferred</center></b><br>"
+	if(!length(voted_for) || poll.allow_revoting)
+		output += "<p><input type='submit' value='Vote'></form>"
+	output += "</div>"
+	src << browse(jointext(output, ""),"window=playerpoll;size=500x500")
 
-//Returns null on failure, TRUE if already voted, FALSE if not voted yet.
-/mob/dead/new_player/proc/poll_check_voted(pollid, text = FALSE, silent = FALSE)
-	var/table = "poll_vote"
-	if (text)
-		table = "poll_textreply"
-	if (!SSdbcore.Connect())
-		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+/**
+  * Runs some poll validation before a vote is processed.
+  *
+  * Checks a player is who they claim to be and that a poll is actually still running.
+  * Also loads the vote_id to pass onto single-option and text polls.
+  * Increments the vote count when successful.
+  *
+  */
+/mob/dead/new_player/proc/vote_on_poll_handler(datum/poll_question/poll, href_list)
+	if(!SSdbcore.Connect())
+		to_chat(src, "<span class='danger'>Failed to establish database connection.</span>")
 		return
-	var/datum/DBQuery/query_hasvoted = SSdbcore.NewQuery("SELECT id FROM `[format_table_name(table)]` WHERE pollid = [pollid] AND ckey = '[ckey]'")
-	if(!query_hasvoted.warn_execute())
-		qdel(query_hasvoted)
+	if(!poll || !href_list)
 		return
-	if(query_hasvoted.NextRow())
-		qdel(query_hasvoted)
-		if(!silent)
-			to_chat(usr, "<span class='danger'>You've already replied to this poll.</span>")
-		return TRUE
-	qdel(query_hasvoted)
-	return FALSE
-
-//Returns adminrank for use in polls.
-/mob/dead/new_player/proc/poll_rank()
-	. = "Player"
-	if(client.holder)
-		. = client.holder.rank.name
-
-
-/mob/dead/new_player/proc/vote_rig_check()
-	if (usr != src)
-		if (!usr || !src)
-			return 0
-		//we gots ourselfs a dirty cheater on our hands!
+	if(usr != src)
+		if(!usr || !src)
+			return
 		log_game("[key_name(usr)] attempted to rig the vote by voting as [key]")
 		message_admins("[key_name_admin(usr)] attempted to rig the vote by voting as [key]")
 		to_chat(usr, "<span class='danger'>You don't seem to be [key].</span>")
 		to_chat(src, "<span class='danger'>Something went horribly wrong processing your vote. Please contact an administrator, they should have gotten a message about this</span>")

```

Comment:
If you are going to move this check to a common proc, you have to protect the `vote_on*` procs from admin calls,

also this was just an older way of checking for admin proc calls before cyberboss build out the current version.

Reply:
Queries created from an advanced proc call are blocked so the `vote_on_` procs are already protected in the sense they can't accomplish anything. I'll add a check anyway as it doesn't hurt to have more safety.

Reply:
Or well, since it's just the same check, short cutting on processing.